feat(api): add project label promotion

StingRayZA opened the pull request:
Adds a mixin that allows the `/promote` endpoint to be called on project labels.

Fixes #1602

JohnVillalovos commented:
Thanks for this. This is going to need some docs updates too, so that people know it is there and also how to use it.

StingRayZA commented:
Sure - will add them now the CI has given the _code_ a 👍

StingRayZA commented:
I've removed the CLI action (It wasn't testing well, and I don't think I know how to make it work without breaking backwards compatibility)

Also added a note into the docs...

StingRayZA commented:
Thanks for the review @nejch - I've got a tweak or two to make and I'll post them ASAP...

nejch commented:
Thanks again @StingRayZA. Just a nitpick for tests and if you could maybe rebase on top of `main`? :)
